src/diffusers/models/transformers/hunyuan_transformer_2d.py

```diff
@@ -0,0 +1,499 @@
+        )
+
+        _, _, height, width = hidden_states.shape
+        height, width = height // self.config.patch_size, width // self.config.patch_size
+
+        # ========================= Build time and image embedding =========================
+        timesteps_projected = self.time_proj(timestep)
+        temb = self.time_embedding(timesteps_projected.to(dtype=timestep.dtype))
+        hidden_states = self.pos_embed(hidden_states)
+
+        # Build image meta size tokens
+        image_meta_size = get_timestep_embedding(image_meta_size.view(-1), 256, True, 0)  # [B * 6, 256]
+
+        image_meta_size = image_meta_size.to(dtype=hidden_states.dtype)
+        image_meta_size = image_meta_size.view(-1, 6 * 256)
+        extra_vec = torch.cat([pooled_projections, image_meta_size], dim=1)  # [B, D + 6 * 256]
+
+        # Build style tokens
+        style_embedding = self.style_embedder(style)
+        extra_vec = torch.cat([extra_vec, style_embedding], dim=1)
+
+        # Concatenate all extra vectors
+        temb = temb + self.extra_embedder(extra_vec)  # [B, D]
+
+        # ========================= Forward pass through HunYuanDiT blocks =========================
+        skips = []
+        for layer, block in enumerate(self.blocks):
+            if layer > self.config.num_layers // 2:
+                skip = skips.pop()
+                hidden_states = block(
+                    hidden_states,
+                    temb=temb,
+                    encoder_hidden_states=encoder_hidden_states,
+                    image_rotary_emb=image_rotary_emb,
+                    skip=skip,
+                )  # (N, L, D)
+            else:
+                hidden_states = block(
+                    hidden_states,
+                    temb=temb,
+                    encoder_hidden_states=encoder_hidden_states,
+                    image_rotary_emb=image_rotary_emb,
+                )  # (N, L, D)
+
+            if layer < (self.config.num_layers // 2 - 1):
+                skips.append(hidden_states)
+
+        # ========================= Final layer =========================
+        hidden_states = self.norm_out(hidden_states, temb.to(torch.float32))
+        hidden_states = self.proj_out(hidden_states)
+        # (N, L, patch_size ** 2 * out_channels)
+
+        # unpatchify: (N, out_channels, H, W)
+        patch_size = self.pos_embed.patch_size
+        hidden_states = hidden_states.reshape(
+            shape=(hidden_states.shape[0], height, width, patch_size, patch_size, self.out_channels)
+        )
+        hidden_states = torch.einsum("nhwpqc->nchpwq", hidden_states)
+        output = hidden_states.reshape(
+            shape=(hidden_states.shape[0], self.out_channels, height * patch_size, width * patch_size)
+        )
+        if not return_dict:
+            return (output,)
+        return Transformer2DModelOutput(sample=output)
+
+    def initialize_weights(self):
```

Comment:
is this needed for training? are we planning to release training script too? 
let's update it if needed, or remove it if not